connector/src/main/java/net/tirasa/connid/bundles/okta/utils/OktaUtils.java

```diff
@@ -39,9 +46,37 @@ public static void handleGeneralError(final String message, final Exception ex)
 
     public static void wrapGeneralError(final String message, final Exception ex) {
         LOG.error(ex, message);
+        if (ex instanceof ResourceException) {
```

Comment:
The only exception that a connector should raise is `ConnectorException`, to make it manageable by the ConnId framework caller.

I'd suggest to adapt the new `wrapResourceException` method to ultimaltely return - rather than throw - an Exception, and to pass such a returned Exception to `ConnectorException#wrap`

Reply:
I changed `wrapResourceException` to return an exception.